packages/core/src/internal/sagaHelpers/debounce.js

```diff
@@ -0,0 +1,41 @@
+import fsmIterator, { qEnd, safeName } from './fsmIterator'
+import { delay, fork, race, take } from '../io'
+import { END } from '../channel'
+
+export default function debounceHelper(delayLength, patternOrChannel, worker, ...args) {
+  let action, raceOutput
+
+  const yTake = {done: false, value: take(patternOrChannel)}
+  const yRace = {
+    done: false,
+    value: race({
+      action: take(patternOrChannel),
+      debounce: delay(delayLength),
+    }),
+  }
+  const yFork = ac => ({done: false, value: fork(worker, ...args, ac)})
+  const yNoop = (value) => ({done: false, value})
+
+  const setAction = ac => action = ac
+  const setRaceOutput = ro => raceOutput = ro
+
+  return fsmIterator(
+    {
+      q1() {
+        return {nextState: 'q2', effect: yTake, stateUpdater: setAction}
+      },
+      q2() {
+        return action === END ?
```

Comment:
I saw that comment https://github.com/redux-saga/redux-saga/pull/688#pullrequestreview-12891349 but haven't checked yet. I'll check the source code later and we can remove those checks from all helpers once

Reply:
I added test for both pattern and channel finished with END, seems that it can be removed across the code base. 
would you mind checking added tests are correct?